pkg/parser/parseable/parseable.go

```diff
@@ -67,5 +69,50 @@ func (parser *DefaultParser) Parse(node *node.Node) error {
 }
 
 func (parser *DefaultParser) Schema() *schema.Schema {
-	return &schema.Schema{}
+	schema := &schema.Schema{
+		Properties:           make(map[string]*schema.Schema),
+		PatternProperties:    make(map[*regexp.Regexp]*schema.Schema),
+		AdditionalItems:      &schema.AdditionalItems{Schema: nil},
+		AdditionalProperties: &schema.AdditionalProperties{Schema: nil},
+	}
+
+	for _, field := range parser.fields {
+		switch nameable := field.name.(type) {
+		case *nameable.SimpleNameable:
+			schema.Properties[nameable.Name()] = field.schema
+		case *nameable.RegexNameable:
+			schema.PatternProperties[nameable.Regex()] = field.schema
+		}
+
+		if field.required && !parser.Collectible() {
+			schema.Required = append(schema.Required, field.name.String())
+		}
+	}
+
+	for _, collectibleField := range parser.collectibleFields {
+		schema.Properties[collectibleField.name] = collectibleField.schema
+	}
+
+	return schema
+}
+
+type DeepField struct {
+	Name   string
+	Schema *schema.Schema
+}
+
+// Note that this is a simplification and it doesn't really return "deep" fields
+// since all of the collectible fields currently are defined on depth 1, which we
+// access here.
+func (parser *DefaultParser) DeepFields() []DeepField {
```

Comment:
Is there a need for such abstraction? Maybe just make collectibleFields public and use it in this one place?

Reply:
>Maybe just make collectibleFields public and use it in this one place?

You can't just make `collectibleFields` fields public since it's hidden behind an interface, which means you'd still have to define a new retrieval method similar to this one in the `Parseable` interface and then implement it.

Reply:
That's OK with me. I just don't like DeepField abstraction that seems was created just for one usage. 🤔

Reply:
Changed in https://github.com/cirruslabs/cirrus-cli/pull/161/commits/37eba8c08904bf0128390541a66d6a81c2ab732d.